lib/mosquitto.h

```diff
@@ -2567,6 +2567,91 @@ libmosq_EXPORT int mosquitto_subscribe_callback(
 		const struct libmosquitto_will *will,
 		const struct libmosquitto_tls *tls);
 
+/*
+ * Function: mosquitto_publish_multiple
+ *
+ * Helper function to make publishing a number of messages very straightforward.
+ *
+ * This connects to a broker, sends the supplied messages, and returns after
+ * disconnecting cleanly.
+ *
+ * Parameters:
+ *   messages - an array of mosquitto_message structs
+ *   message_count - the number of messages supplied
+ *   host - the broker to connect to.
+ *   port - the network port the broker is listening on.
+ *   client_id - the client id to use, or NULL if a random client id should be
+ *               generated.
+ *   keepalive - the MQTT keepalive value.
+ *   clean_session - the MQTT clean session flag.
+ *   username - the username string, or NULL for no username authentication.
+ *   password - the password string, or NULL for an empty password.
+ *   will - a libmosquitto_will struct containing will information, or NULL for
+ *          no will.
+ *   tls - a libmosquitto_tls struct containing TLS related parameters, or NULL
+ *         for no use of TLS.
+ *
+ * Returns:
+ *   MOSQ_ERR_SUCCESS - on success
+ *   >0 - on error.
+ */
+libmosq_EXPORT int mosquitto_publish_multiple(
+		struct mosquitto_message **messages,
```

Comment:
Who owns **messages after this is called? It seems like the function doesn't modify the messages, so they should be declared as `struct mosquitto_message *const *const ` please.